### PR TITLE
V2 sentinel 2 date search

### PR DESCRIPTION
**Pull request by pablotcarreira**
Those changes allow the user to restrict the search by a date interval.
The start_date and end_date are optional. If no date is defined the function will search images between 2015 and now.

### CODE REVIEW

File: tests/fixtures/expected_results_sentinel2.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
...
+
+
+filter = [{'sat': 'S2A', 'path': 'tiles/22/K/HV/2017/1/12/0/', 'utm_zone': '22', 'latitude_band': 'K', 'grid_square': 'HV', 'num': '0', 'acquisition_date': '20170112', 'browseURL': 'https://sentinel-s2-l1c.s3.amazonaws.com/tiles/22/K/HV/2017/1/12/0/preview.jpg', 'scene_id': 'S2A_tile_20170112_22KHV_0'}, {'sat': 'S2A', 'path': 'tiles/22/K/HV/2
```
Comment (vincentsarago) [inline tests/fixtures/expected_results_sentinel2.py]: @pablotcarreira is this used somewhere ?

File: tests/test_search.py [hunk 2]
```diff
@@ -455,3 +456,10 @@ def test_cbers_awfi_valid(list_directory, session):
         'version': '4'}]
 
     assert list(search.cbers(path, row, sensor)) == expected
+
+
+def test_s2_date_filter():
+    start_date = datetime(2017, 1, 1)
+    end_date = datetime(2017, 5, 15)
+    results_date_filter = list(search.sentinel2(22, "K", "HV", start_date=start_date, end_date=end_date))
+    assert len(results_date_filter) == 22
```
Comment (vincentsarago) [inline tests/test_search.py]: @pablotcarreira Can you add some tests for https://github.com/RemotePixel/aws-sat-api-py/pull/6/files#diff-fc9a6edf050c18c92ddbbaa9b1ccefc0R179 and https://github.com/RemotePixel/aws-sat-api-py/pull/6/files#diff-fc9a6edf050c18c92ddbbaa9b1ccefc0R182

File: tests/timing.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+def timing_func():
+    start_date = datetime(2017, 1, 1)
+    end_date = datetime(2017, 5, 15)
+    sentinel2(22, "K", "HV", start_date=start_date, end_date=end_date)
+
+
+if __name__ == '__main__':
+    t = timeit.timeit('timing_func()', number=10, setup="from __main__ import timing_func")
+    print(t)
```
Comment (vincentsarago) [inline tests/timing.py]: @pablotcarreira What is this script for ? I don't see it used ?

File: tests/fixtures/expected_results_sentinel2.py (hunk 1 above)
Comment (pablotcarreira) [reply tests/fixtures/expected_results_sentinel2.py]: It was the remnant of an ancient methodology. I removed it.

**Comment by pablotcarreira** [comment]
tests/timing.py was used during development to test the execution time. I removed it now.

File: tests/timing.py (hunk 3 above)
Comment (pablotcarreira) [reply tests/timing.py]: tests/timing.py was used during development to test the execution time. I removed it now.

**Comment by pablotcarreira** [comment]
Thank you for the changes. I did not know how to do the tests without a real call to AWS, It's a new trick and I'm grateful to learn. ;)

**Comment by vincentsarago** [comment]
Yes mocking the multithreading calls is not a easy task, that's why at first I wasn't testing the `sentinel2` search function :-)

**Pull request merged**